tools/buildtools/packmol/packmol.xml

```diff
@@ -0,0 +1,196 @@
+<tool id="packmol" name="PACKMOL" version="1.18.013">
+    <description>Initial configurations for Molecular Dynamics Simulations by packing optimization</description>
+    <macros>
+        <import>macros_packmol.xml</import>
+    </macros>
+    <requirements>
+      <requirement type="package" version="1!18.013">packmol</requirement>
+    </requirements>
+    <command detect_errors="exit_code">
+        <![CDATA[
+        python "$packmol_script" '$inputs' 
+        ]]>
+    </command>
+    <configfiles>
+        <inputs name="inputs" />
+        <configfile name="packmol_script">
+            <![CDATA[
+import sys
+import json
+
+
+input_json_path = sys.argv[1]
+params = json.load(open(input_json_path, "r"))
+
+# get the inputs
+input_iter = []
+#for $i, $s in enumerate( $packmol_format.packmol_inputs )
+pmfile_index = $i 
+pmfile_path = "${s.input.file_name}"
+input_iter.append(pmfile_path)
+params["packmol_format"]["packmol_inputs"][pmfile_index]["file"] = pmfile_path
+params["packmol_format"]["packmol_inputs"][pmfile_index]["datasetname"] = "${s.input.name}"
+#end for        
+
+params["outfile"] = "$outfile" 
+
+import os
+from jinja2 import Environment, FileSystemLoader
+
+currentpath = "$__tool_directory__"  # should work generally
+template_environment = Environment(loader=FileSystemLoader(currentpath),lstrip_blocks=True, trim_blocks=True)
+template = template_environment.get_template('template.j2')
+print(params)
+
+with open("$parameteroutfile",'w+') as f:
+    f.write(template.render(params))
+
+os.system("packmol < "+"$parameteroutfile")
```

Comment:
os.system is no really recommend to used theses days. I think you can remove this call here and put this straight as `packmol < '$parameteroutfile'` into the cheetah-command above.